node/pkg/watchers/aptos/watcher.go

```diff
@@ -82,16 +93,26 @@ func (e *Watcher) Run(ctx context.Context) error {
 
 	logger := supervisor.Logger(ctx)
 
-	logger.Info("Starting watcher",
-		zap.String("watcher_name", e.networkID),
-		zap.String("rpc", e.aptosRPC),
-		zap.String("account", e.aptosAccount),
-		zap.String("handle", e.aptosHandle),
-	)
+	if e.useIndexer {
+		logger.Info("Starting Aptos watcher in indexer mode",
+			zap.String("watcher_name", e.networkID),
+			zap.String("rpc", e.aptosRPC),
+			zap.String("account", e.aptosAccount),
+			zap.String("handle", e.aptosHandle),
+			zap.String("indexerRpc", e.aptosIndexerRPC),
+		)
+	} else {
+		logger.Info("Starting Aptos watcher in legacy mode",
+			zap.String("watcher_name", e.networkID),
+			zap.String("rpc", e.aptosRPC),
+			zap.String("account", e.aptosAccount),
+			zap.String("handle", e.aptosHandle),
+		)
+	}
 
 	// SECURITY: the API guarantees that we only get the events from the right
 	// contract
-	var eventsEndpoint = fmt.Sprintf(`%s/v1/accounts/%s/events/%s/event`, e.aptosRPC, e.aptosAccount, e.aptosHandle)
+	var eventsEndpoint = e.aptosIndexerRPC
```

Comment:
nit: It looks like these two variables are just used for the log statement below, so you could consider in-lining them.